handyrl/worker.py

```diff
@@ -160,9 +157,25 @@ def run(self):
                     self.result_send_cnt = 0
 
 
-def gather_loop(args, conn, gaid):
+def gather_loop(args, conn, gather_id):
+    n_pro, n_ga = args['worker']['num_parallel'], args['worker']['num_gathers']
+    n_pro_w = (n_pro // n_ga) + int(gather_id < n_pro % n_ga)
+    args['worker']['num_parallel_per_gather'] = n_pro_w
+    base_worker_id = 0
+
+    if conn is None:
+        # entry
+        conn = connect_socket_connection(args['worker']['server_address'], 9998)
+        conn.send(args['worker'])
+        args = conn.recv()
+
+        if gather_id == 0:  # call once at every machine
+             print(args)
+             prepare_env(args['env'])
```

Comment:
Is this safe...? I can't think of any problems though...

Reply:
I'd like to remove prepare_env() ...